parser/parser.go

```diff
@@ -428,7 +479,7 @@ func ParseAggregate(expr sqlparser.Expr) (logical.Aggregate, logical.NamedExpres
 				return "", nil, errors.Wrap(err, "couldn't parse aggregate argument")
 			}
 
-		case *sqlparser.StarExpr:
+		case *sqlparser.StarExpr: //TODO: not sure this is reachable tbh
```

Comment:
Isn't it the case for COUNT(*) or are they handled differently?

At least panic, and don't leave a mine like parseArg = nil (I think it's a mine, isn't it?)

Reply:
Oh it actually is reachable, and the nil is because when you return from it, if the result is nil you know you are aggregating stars